### PR TITLE
Add org.ryujinx.Ryujinx

### PR DESCRIPTION
**Pull request by albfan**
nintendo switch emulator build in .Net https://ryujinx.org/

After fetch the dependencies with flatpak-dotnet-generator.py and remove the test projects (seems the generator is not retrieving test dependencies for some reason)

dotnet sdk is installed and project is build from flatpak

Running it there's an error:

```
00:00:01.119 |N| Application : Unhandled exception caught: System.InvalidCastException: Unable to cast object of type 'OpenTK.Platform.X11.X11WindowInfo' to type 'OpenTK.Platform.Linux.LinuxWindowInfo'.
   at OpenTK.Platform.Linux.LinuxFactory.CreateGLContext(GraphicsMode mode, IWindowInfo window, IGraphicsContext shareContext, Boolean directRendering, Int32 major, Int32 minor, GraphicsContextFlags flags)
   at OpenTK.Graphics.GraphicsContext..ctor(GraphicsMode mode, IWindowInfo window, IGraphicsContext shareContext, Int32 major, Int32 minor, GraphicsContextFlags flags)
   at OpenTK.Graphics.GraphicsContext..ctor(GraphicsMode mode, IWindowInfo window, Int32 major, Int32 minor, GraphicsContextFlags flags)
   at OpenTK.GLWidget.Initialize()
   at OpenTK.GLWidget.OnDrawn(Context cr)
   at Gtk.Widget.Drawn_cb(IntPtr inst, IntPtr cr)
```

If that can be resolved I will update appdata info, etc...

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.github.Ryujinx

File: com.github.Ryujinx.json
Comment (LeandroStanger) [inline com.github.Ryujinx.json:53]:
```suggestion
                    "type": "git",
                    "url": "https://github.com/Ryujinx/Ryujinx"
                    "commit": "09341dc11d88de39b2b4fb990ec730b4d57f8b0a"
                },
```

**Comment by albfan** [comment]
@thog I modify the opentk to use the one from Ryujinx. Should I build from master? seems the last release was 1.0.2 which leads to this errors:

```
/run/build/ryujinx/Ryujinx.HLE/Ryujinx.HLE.csproj : error NU1102: Unable to find package OpenTK.NetStandard with version (>= 1.0.5.12) [/run/build/ryujinx/Ryujinx.sln]
/run/build/ryujinx/Ryujinx.HLE/Ryujinx.HLE.csproj : error NU1102:   - Found 1 version(s) in /run/build/ryujinx/nuget-sources [ Nearest version: 1.0.2 ] [/run/build/ryujinx/Ryujinx.sln]
```

**Comment by LeandroStanger** [comment]
bot, build com.github.Ryujinx

**Comment by LeandroStanger** [comment]
bot, build com.github.Ryujinx

**Comment by albfan** [comment]
Just removed the custom opentk, doesn't seems to fix anything and build on host relays on standard one

**Comment by albfan** [comment]
bot, build com.github.Ryujinx

File: com.github.Ryujinx.json
Comment (LeandroStanger) [inline com.github.Ryujinx.json:4]: Update runtime to 20.08
```suggestion
    "runtime-version": "20.08",
```

File: com.github.Ryujinx.json
Comment (LeandroStanger) [inline com.github.Ryujinx.json:53]:
```suggestion
		    "commit": "4f3ae6f62ca5858c2b74eba5a15bad5dff580349"
```

File: com.github.Ryujinx.json
Comment (barthalion) [inline com.github.Ryujinx.json:59]: Any chance you could move generated part to `generated-sources.json`?

File: com.github.Ryujinx.json
Comment (barthalion) [reply com.github.Ryujinx.json:4]: @LeandroStanger You haven't tested your change, have you? There's no dotnet extension for 20.08 yet. Please don't suggest anything you did not try.

File: com.github.Ryujinx.json
Comment (barthalion) [inline com.github.Ryujinx.json:13]: Does it actually work natively on Wayland?

File: com.github.Ryujinx.json
Comment (barthalion) [inline com.github.Ryujinx.json:14]: Does it need full host access?

File: com.github.Ryujinx.json
Comment (barthalion) [inline com.github.Ryujinx.json:2]: The project has a domain so something like `org.ryujinx.Ryujinx` would be a better fit.

**Review by barthalion — CHANGES_REQUESTED** [review]
Judging by the build error, it works only on x86_64. Please add flathub.json as described at https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

Have you contacted upstream team about maintaining their app officially here?

File: com.github.Ryujinx.json
Comment (albfan) [reply com.github.Ryujinx.json]: It needs access to several directories for data, but they are configurable. Should I remove this?

File: com.github.Ryujinx.json
Comment (albfan) [reply com.github.Ryujinx.json]: No

File: com.github.Ryujinx.json
Comment (albfan) [reply com.github.Ryujinx.json]: Still not dotnet extension

File: com.github.Ryujinx.json
Comment (albfan) [reply com.github.Ryujinx.json]: Much cleaner!

File: com.github.Ryujinx.json
Comment (albfan) [reply com.github.Ryujinx.json]: Yep

**Comment by albfan** [comment]
bot, build com.github.Ryujinx

**Comment by bilelmoussaoui** [comment]
bot, build org.ryujinx.Ryujinx

**Comment by albfan** [comment]
Looks like adding generated-sources.json on a json manifest fails to correctly fetch the sources.  Is it a known bug?

**Comment by albfan** [comment]
bot, build org.ryujinx.Ryujinx

**Comment by albfan** [comment]
bot, build org.ryujinx.Ryujinx

**Comment by albfan** [comment]
bot, build org.ryujinx.Ryujinx

**Comment by albfan** [comment]
bot, build org.ryujinx.Ryujinx

**Comment by albfan** [comment]
build is succesful, but error running remains:

```
00:00:00.957 |N| Application : Unhandled exception caught: System.InvalidCastException: Unable to cast object of type 'OpenTK.Platform.X11.X11WindowInfo' to type 'OpenTK.Platform.Linux.LinuxWindowInfo'.
   at OpenTK.Platform.Linux.LinuxFactory.CreateGLContext(GraphicsMode mode, IWindowInfo window, IGraphicsContext shareContext, Boolean directRendering, Int32 major, Int32 minor, GraphicsContextFlags flags)
   at OpenTK.Graphics.GraphicsContext..ctor(GraphicsMode mode, IWindowInfo window, IGraphicsContext shareContext, Int32 major, Int32 minor, GraphicsContextFlags flags)
   at OpenTK.Graphics.GraphicsContext..ctor(GraphicsMode mode, IWindowInfo window, Int32 major, Int32 minor, GraphicsContextFlags flags)
   at OpenTK.GLWidget.Initialize()
   at OpenTK.GLWidget.OnDrawn(Context cr)
   at Gtk.Widget.Drawn_cb(IntPtr inst, IntPtr cr)
```

**Comment by barthalion** [comment]
It seems to start fine here, but I'm not particularly interested in dumping keys from my Switch to test it further.

File: org.ryujinx.Ryujinx.json
Comment (barthalion) [inline org.ryujinx.Ryujinx.json:13]: If configuration follows XDG spec then `--filesystem=home` and `--filesystem=/run/media' would be still better.

**Comment by barthalion** [comment]
Have you contacted upstream team about maintaining their app officially here?

**Comment by albfan** [comment]
@barthalion yes, it fails starting a game.
I talked with maintainers in his discord server. Is there any other option than maintain the sources here?

**Comment by barthalion** [comment]
Unless upstream team insists on re-using their own pipeline, the repo here is preferred. I can imagine that the manifest could just re-use the binary build available on website? Maintainers of OpenRCT2 are working on triggering update based on external CI at https://github.com/flathub/io.openrct2.OpenRCT2/pull/5, there's also https://github.com/flathub/flatpak-external-data-checker/ .

**Comment by albfan** [comment]
@barthalion as running has problems I tried too using the provided binaries from ryujinxs, but they use a path that leads to ro filesystem so I have to patch the code.

Until that is merged upstream we need to build the code.

I think they are fine with build from here

File: org.ryujinx.Ryujinx.json
Comment (LeandroStanger) [inline org.ryujinx.Ryujinx.json:1286]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by LeandroStanger** [comment]
bot, build org.ryujinx.Ryujinx

File: org.ryujinx.Ryujinx.json
Comment (albfan) [reply org.ryujinx.Ryujinx.json:1286]: Not sure what metainfo is

Comment (bilelmoussaoui) [reply org.ryujinx.Ryujinx.json:1286]: Newer appdata files should use .metainfo.xml extension instead & installed under /metainfo instead of /appdata. See the specs

Comment (albfan) [reply org.ryujinx.Ryujinx.json:1286]: thanks Bilel, TL;DR; hehehe

File: org.ryujinx.Ryujinx.json
Comment (LeandroStanger) [inline org.ryujinx.Ryujinx.json:53]: It should use full commit hash.
```suggestion
                    "commit": "e7d09a8080f6a99b099f2b36f8a26a0c6a89cef7"
```

**Comment by LeandroStanger** [comment]
bot, build org.ryujinx.Ryujinx

**Comment by barthalion** [comment]
@albfan Please poke us when the issue is fixed and it can be merged. Also who from upstream should be added to the new repo?

**Comment by albfan** [comment]
@emmauss might now an ETA for that. @thog should know better who to add.

**Comment by albfan** [comment]
@refi64 found the solution.

@barthalion the way opentk detects x11 running needs a libX11.so and we only have libX1.so.whatever, so it fails. It is fixed.

@LeandroStanger I don't move to master because it needs another patch to be able to read/write some config files. @Thog do you think those patches can land on upstream?

**Comment by albfan** [comment]
Sadly, looks like execution renders graphics worse than run on host

**Comment by LeandroStanger** [comment]
bot, build org.ryujinx.Ryujinx

File: org.ryujinx.Ryujinx.json
Comment (LeandroStanger) [inline org.ryujinx.Ryujinx.json:1287]:
```suggestion
                    "path": "org.ryujinx.Ryujinx.metainfo.xml"
```

File: org.ryujinx.Ryujinx.json
Comment (LeandroStanger) [inline org.ryujinx.Ryujinx.json:54]:
```suggestion
                    "commit": "f89b754abb3d721c09ba6ab00f7906d2d5ccb02b"
```

File: org.ryujinx.Ryujinx.json
Comment (albfan) [reply org.ryujinx.Ryujinx.json]: done

File: org.ryujinx.Ryujinx.json
Comment (albfan) [reply org.ryujinx.Ryujinx.json:54]: Move needs additional patches about starting the app (which now reads arguments and tries to write in /app directory

**Comment by nedrichards** [comment]
bot, build org.ryujinx.Ryujinx

**Comment by anthr76** [comment]
Is this okay to be merged now?

**Comment by albfan** [comment]
No there're unresolved conversation about new patches needed for master (more  directories created on working dir that will not work on flatpak sandbox)

**Comment by LeandroStanger** [comment]
@albfan Any progress?

File: org.ryujinx.Ryujinx.json
Comment (LeandroStanger) [inline org.ryujinx.Ryujinx.json:4]: Update runtime to 20.08

**Comment by albfan** [comment]
Have to review all again.

There's a rewrite about opentk and looks like paths needs to be patched for flatpak.

File: org.ryujinx.Ryujinx.json
Comment (hfiguiere) [inline org.ryujinx.Ryujinx.json:64]: can't we just include the generated sources json here? this would make the manifest easier to maintain

File: flathub.json
Comment (hfiguiere) [inline flathub.json:2]: since it is using dotnet and the runtime exists for aarch64, is there a reason to limit to x86_64?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by Somaxa8** [comment]
Hi, how is the status of this? I have seen freedesktop sdk 21.08 has been released recently.

**Comment by albfan** [comment]
I see the branch name is incorrect. There's work to do to bundle this, and upstream project has change a bit.

Work left on https://github.com/albfan/flathub/tree/Ryujinx

**Comment by barthalion** [comment]
It's just a branch name. It's not relevant to anything.

**Comment by albfan** [comment]
funnily enough my local branch new-pr is the base for other PR. A problem with a simple fix (add upstream repo and refer to upstream/new-pr)

Just trying to stay as simple as possible.

Ryujinx uses an automatic download to update app, so bundle on flatpak will get quickly out of date.

Still lot of config files are created on current run dir, which goes to /app/bin dir. No plans upstream to fix that

**Comment by iMonZ** [comment]
anything new here?
I would like to try out the Ryujinx emulator but without this issue closed it wont happen :(

**Pull request closed without merging**
